test/dhcp_libvirt/dhcp_libvirt_provider_test.rb

```diff
@@ -0,0 +1,89 @@
+require 'test_helper'
+require 'dhcp_libvirt/dhcp_libvirt'
+require 'dhcp_libvirt/dhcp_libvirt_main'
+require 'dhcp_common/dependency_injection/dependencies'
+
+class DhcpLibvirtProviderTest < Test::Unit::TestCase
+  def setup
+    @service = Proxy::DHCP::SubnetService.new
+    @subnet_store = @service.subnets = Proxy::MemoryStore.new
+    @subject = ::Proxy::DHCP::Libvirt::Provider.new.initialize_for_testing(
```

Comment:
A bunch of tests fail here, as the constructor instantiates Proxy::LibvirtNetwork, which opens a connection to libvirt. Same issue in dns tests.